### PR TITLE
Fixes slow circuit cloning breaking with some circuits, and makes it use world time checks instead of decrementing countdowns

### LINKED ISSUE #36239: Circuit printer says there's negative time until print is completed
1. take circuit imprinter
2. insert metal
3. i pasted in this script: https://tgstation13.org/phpBB/viewtopic.php?f=2&t=14420#p365647
4. start printing it
5. when the timer hits 0 it keeps going well into the negatives. print never completes

apparently my circuit printer wasnt upgraded or something maybe thats part of it 
round ID 84620 on bagil

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
fix: Circuit slow-cloning no longer breaks with some circuits.
code: Circuit slow-cloning is now cleaner.
/:cl:

Fixes #36239.

Circuit cloning now uses a `world.time` check to print its circuit, instead of manually decrementing a countdown var and spawning in the circuit once it hit zero. This also swaps out all messages related to that variable with their appropriate `DisplayTimeText()` calls.

### CODE REVIEW

File: code/modules/integrated_electronics/core/printer.dm [hunk 1]
```diff
@@ -273,11 +272,11 @@
...
 						to_chat(usr, "<span class='warning'>You need [program["metal_cost"]] metal to build that!</span>")
 						return
-					var/cloning_time = program["metal_cost"] / 150
+					var/cloning_time = round(program["metal_cost"] / 15)
 					cloning_time = min(cloning_time, MAX_CIRCUIT_CLONE_TIME)
 					cloning = TRUE
-					clone_countdown = cloning_time
-					to_chat(usr, "<span class='notice'>You begin printing a custom assembly. This will take approximately [round(cloning_time / 60, 0.1)] minute(s). You can still print \
+					clone_countdown = world.time + cloning_time
+					to_chat(usr, "<span class='notice'>You begin printing a custom assembly. This will take approximately [DisplayTimeText(cloning_time)]. You can still print \
```
Comment (duncathan) [inline code/modules/integrated_electronics/core/printer.dm]: should this also be `cloning_time + world.time`?

Comment (duncathan) [reply code/modules/integrated_electronics/core/printer.dm]: err, minus not plus

Comment (Xhuis) [reply code/modules/integrated_electronics/core/printer.dm]: It didn't seem to be the case in testing. I'm not in a spot to test it right now though, so it's your call.

Comment (duncathan) [reply code/modules/integrated_electronics/core/printer.dm]: on another glance, I misread. confused the two cloning vars.

**Pull request merged**